### PR TITLE
Revert "Use flagext module from grafana/dskit (#4443)"

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does**:
Revert usage of grafana/dskit/flagext package. I've copied the current flagext code from dskit.

It's not necessary to add a changelog entry, right?

**Which issue(s) this PR fixes**:

**Checklist**
- [x] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
Thank you.

**Comment by aknuds1** [comment]
Integration test failure looks like a flake.

File: pkg/util/flagext/deprecated.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// DeprecatedFlagsUsed is the metric that counts deprecated flags set.
+var DeprecatedFlagsUsed = promauto.NewCounter(
+	prometheus.CounterOpts{
+		Namespace: "cortex",
```
[the remark this answers is not in the document]
Comment (aknuds1) [reply pkg/util/flagext/deprecated.go:15]: Thanks @pracucci, this was what I was meaning to review again myself :)

[the remark this answers is not in the document]
Comment (aknuds1) [reply pkg/util/flagext/deprecated.go]: @pracucci removed the namespace again.

**Comment by aknuds1** [comment]
The tests fail due to a conflict between metrics named the same in dskit/flagext and pkg/util/flagext, likely because dskit/flagext is used internally in dskit. Might have to drop other parts of dskit first in order to resolve.

**Review by pracucci — APPROVED** [review]
Thanks!

**Pull request merged**